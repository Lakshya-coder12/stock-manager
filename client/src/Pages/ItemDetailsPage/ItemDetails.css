@media (max-width: 480px) {
  .item-details {
    width: 75vw;
    border-radius: 5px;
    padding: 10px;
    background-color: white;
    box-shadow: 3px 3px #babab8;
    margin: 0 auto;
    margin-top: 100px;
  }
}

@media (min-width: 480px) and (max-width: 950px) {
  .item-details {
    width: 50vw;
    border-radius: 5px;
    padding: 10px;
    background-color: white;
    box-shadow: 3px 3px #babab8;
    margin: 0 auto;
    margin-top: 100px;
  }
}
@media (min-width: 950px) {
  .item-details {
    width: 30vw;
    border-radius: 5px;
    padding: 10px;
    background-color: white;
    box-shadow: 3px 3px #babab8;
    margin: 0 auto;
    margin-top: 100px;
  }
}

.item {
  margin-top: 15px;
}

.value {
  padding-top: 10px;
}

.item-title {
  color: #1976c2;
  font-weight: 600;
  font-size: large;
}

.form-btn {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}
