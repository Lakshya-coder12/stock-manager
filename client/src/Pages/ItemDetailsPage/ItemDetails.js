import React, { Component } from "react";
import { Link } from "react-router-dom";
import Button from "../../Components/Button/Button";
import "./ItemDetails.css";
import moment from "moment";
import PropTypes from "prop-types";
// import { connect } from "react-redux";
// import {
//   deleteItem,
//   fetchSingleItem,
//   removeSingleItem,
// } from "../../redux/actions/ItemActions";
import jwt_decode from "jwt-decode";

class ItemDetails extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
    };
  }
  itemID = this.props.match.params;
  componentDidMount() {
    const token = localStorage.getItem("jwt");
    if (!token) {
      this.props.history.push("/login");
    } else {
      const user = jwt_decode(token);
      const username = user.user.name;
      this.setState({
        username: username,
      });
      this.props.fetchSingleItem(this.itemID.id);
    }
  }
  handleRedirect = () => {
    this.props.history.push("/home");
  };
  componentWillUnmount() {
    this.props.removeSingleItem();
  }

  handleLogout = () => {
    this.props.logout(this.props.history);
  };

  handleDeleteItem = () => {
    this.props.deleteItem(this.itemID.id, this.props.history);
  };
  render() {
    const { loading, item } = this.props.item;
    if (loading) {
      return <div>....loading</div>;
    } else {
      return (
        <>
          <Button className="abs-button" onClick={this.handleRedirect}>
            Home
          </Button>
          <div className="item-details">
            <div className="item">
              <div className="item-title">Name</div>

              <span className="value">{item && item.name}</span>
            </div>
            <div className="item">
              <div className="item-title">Company Name</div>
              <span className="value">{item && item.companyName}</span>
            </div>
            <div className="item">
              <div className="item-title">Stock</div>
              <span className="value">{item && item.stock}</span>
            </div>
            <div className="item">
              <div className="item-title">Date Created</div>
              <span className="value">
                {moment(item.dateCreated).toDate().toDateString()}
              </span>
            </div>
            <div className="form-btn">
              <Link to={`/edit/${this.itemID.id}`}>
                <Button>Edit Item</Button>
              </Link>
              <Button onClick={this.handleDeleteItem}>Delete Item</Button>
            </div>
          </div>
          <div className="logout">
            <span className="username">{this.state.username}</span>
            <Button onClick={this.handleLogout}>Logout</Button>
          </div>
        </>
      );
    }
  }
}

ItemDetails.propTypes = {
  fetchSingleItem: PropTypes.func,
  removeSingleItem: PropTypes.func,
  deleteItem: PropTypes.func,
  logout: PropTypes.func,
};

// const mapStateToProps = (state) => {
//   return {
//     item: state.singleItem,
//   };
// };

// const mapDispatchToProps = (dispatch) => {
//   return {
//     fetchSingleItem: (id) => dispatch(fetchSingleItem(id)),
//     removeSingleItem: () => dispatch(removeSingleItem()),
//     deleteItem: (id, history) => dispatch(deleteItem(id, history)),
//   };
// };

// export default connect(mapStateToProps, mapDispatchToProps)(ItemDetails);
export default ItemDetails;
