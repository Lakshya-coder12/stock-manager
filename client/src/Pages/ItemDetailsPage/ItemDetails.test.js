import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import ItemDetails from "./ItemDetails";

configure({ adapter: new Adapter() });

describe("Testing Item Details page", () => {
  it("matches snapshot", () => {
    const paramsMock = jest.fn();
    const pushMock = jest.fn();
    const itemObject = {
      item: {
        name: "Shoes",
        companyName: "Nike",
        stock: 100,
        dateCreated: 1647519150858.0,
      },
    };
    const fetchSingleItemMock = jest.fn();
    const deleteItemMock = jest.fn();
    const removeSingleItemMock = jest.fn();
    const wrapper = shallow(
      <ItemDetails
        item={itemObject}
        fetchSingleItem={fetchSingleItemMock}
        deleteItem={deleteItemMock}
        removeSingleItem={removeSingleItemMock}
        match={{ params: paramsMock }}
        history={{ push: pushMock }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
