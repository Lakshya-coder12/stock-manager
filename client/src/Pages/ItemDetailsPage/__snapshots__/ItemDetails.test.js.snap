// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Item Details page matches snapshot 1`] = `
<Fragment>
  <Button
    className="abs-button"
    onClick={[Function]}
  >
    Home
  </Button>
  <div
    className="item-details"
  >
    <div
      className="item"
    >
      <div
        className="item-title"
      >
        Name
      </div>
      <span
        className="value"
      >
        Shoes
      </span>
    </div>
    <div
      className="item"
    >
      <div
        className="item-title"
      >
        Company Name
      </div>
      <span
        className="value"
      >
        Nike
      </span>
    </div>
    <div
      className="item"
    >
      <div
        className="item-title"
      >
        Stock
      </div>
      <span
        className="value"
      >
        100
      </span>
    </div>
    <div
      className="item"
    >
      <div
        className="item-title"
      >
        Date Created
      </div>
      <span
        className="value"
      >
        Thu Mar 17 2022
      </span>
    </div>
    <div
      className="form-btn"
    >
      <Link
        to="/edit/undefined"
      >
        <Button>
          Edit Item
        </Button>
      </Link>
      <Button
        onClick={[Function]}
      >
        Delete Item
      </Button>
    </div>
  </div>
  <div
    className="logout"
  >
    <span
      className="username"
    />
    <Button
      onClick={[Function]}
    >
      Logout
    </Button>
  </div>
</Fragment>
`;
